src/react/boosts/BoostForm.tsx

```diff
@@ -0,0 +1,241 @@
+
+                  <FormTitle>End date</FormTitle>
+                  <input type="date" name="dateEnd" value={this.state.dateEndInput} onChange={this.onDateEndInputChange} />
+
+                  <FormTitle>Newsroom Wallet</FormTitle>
+                  <TextInput name="newsroomWallet" value={this.props.newsroomWallet} disabled />
+
+                  <Button size={buttonSizes.MEDIUM} type="submit" disabled={this.state.loading || !!this.state.boostId}>Save Boost</Button>
+                  <div>
+                    {this.state.loading && "loading..."}
+                    {this.state.error && (<Error>{this.state.error}</Error>)}
+                    {this.state.boostId && ("Boost created, ID: " + this.state.boostId)}
+                  </div>
+                </BoostWrapper>
+              </form>
+            );
+          }}
+        </Mutation>
+      </>
+    );
+  }
+
+  private onInputChange = (name: string, val: string) => {
+    this.setState({
+      boost: {
+        ...this.state.boost,
+        [name]: val,
+      },
+    })
+  };
+
+  private onDateEndInputChange = (event: any) => {
+    event.preventDefault();
+    this.setState({
+      dateEndInput: event.target.value,
+      boost: {
+        ...this.state.boost,
+        dateEnd: new Date(event.target.value),
+      },
+    })
+  };
+
+  private onItemInputChange = (i: number, name: "cost" | "item", val: string) => {
+    const items = this.state.boost.items!;
+
+    let goalAmount = this.state.boost.goalAmount || 0;
+    if (name === "cost") {
+      const oldVal = items[i][name] || 0;
+      items[i][name] = parseFloat(val);
+      goalAmount = goalAmount - oldVal + items[i][name];
+    } else {
+      items[i][name] = val;
+    }
+
+    this.setState({
+      boost: {
+        ...this.state.boost,
+        goalAmount,
+        items,
+      },
+    });
+  };
+
+  private addItem = (event: React.MouseEvent) => {
+    event.preventDefault();
+    this.setState({
+      boost: {
+        ...this.state.boost,
+        items: this.state.boost.items!.concat({} as any),
+      },
+    });
+  }
+
+  private removeItem = (event: React.MouseEvent, i: number) => {
+    event.preventDefault();
+    this.setState({
+      boost: {
+        ...this.state.boost,
+        goalAmount: (this.state.boost.goalAmount || 0) - (this.state.boost.items![i].cost || 0),
+        items: this.state.boost.items!.slice(0, i).concat(this.state.boost.items!.slice(i+1)),
+      },
+    });
+  }
+
+  private async handleSubmit(event: React.FormEvent, mutation: MutationFunc): Promise<void> {
+    event.preventDefault();
+    const boost = this.state.boost;
+    this.setState({ loading: true, error: undefined });
+    try {
+      const response = await mutation({
+        variables: {
+          input: {
+            channelID: this.props.newsroomAddress, // @TODO/tobek Is this the right field to use?
```

Comment:
this should almost certainly be the newsroom owner's address (e.g. the multisig)

Reply:
@dankins can you confirm what `channelID` should be for a boost?

Reply:
i was thinking that this would be something like @user, %newsroom, #topic and they could all be channels, which would refer to a `channels` table (and id). until we sort that out let's just use the newsroom smart contract address